nixos/modules/system/activation/activation-script.nix

```diff
@@ -154,9 +154,15 @@ in
 
     system.activationScripts.tmpfs =
       ''
-        ${pkgs.utillinux}/bin/mount -o "remount,size=${config.boot.devSize}" none /dev
```

Comment:
Also drops options when given just a device. So to preserve all the options you need to avoid mentioning both device and filesystem type.